script.js

```diff
@@ -1,4 +1,204 @@
+
+// Track game mode - allows the game to move into the different player stages eg. rolling dice, getting sequence etc
+var gameMode = 1;
+
+// Variable to store a player's dice rolls
+var playerRoll = [];
+
+// Variable to store all player's combined numbers
+var playerCombinedNums = [];
+
+// Variable to track running scores of each player.
+var runningScore = [];
+
+// Function to get user inout for game version.
+var getVersion = function (input) {
+  gameVersion = input;
+  return `In this mode, the ${gameVersion} number will win the game. <br><br> Please input the number of players.`;
+};
+
+// Function to generate dice roll
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//Request for player input to get number of players in game.
+var getPlayers = function (input) {
+  numPlayers = input;
+  return `Welcome! There are ${numPlayers} players in this game. <br><br> Please select how many dice you want to play with. `;
+};
+
+// Get player input to number of dice used in game.
+var getNumDice = function (input) {
+  numDice = input;
+  return `You will be playing with ${numDice} dice in this game. <br><br> Player 1, please click "submit" to roll the dice.`;
+};
+
+// Get player to roll dice.
+var getPlayerRoll = function () {
+  var index = 0;
+  var playerDiceRoll = [];
+  while (index < numDice) {
+    playerDiceRoll.push(diceRoll());
+    index += 1;
+  }
+  return playerDiceRoll;
+};
+
+// Get player to select the sequence of the dice. Function takes in the sequence input, and the array of dice rolls.
+var getSequence = function (input, diceArray) {
+  var combinedNum = 0;
+  var autoCombinedNum = 0;
+  var indexArray = input.split("").map(Number);
+  // This loop takes in the user input sequence and generate the combined number.
+  for (var i = 0; i < indexArray.length; i += 1) {
+    combinedNum = combinedNum * 10 + diceArray[indexArray[i]];
+  }
+  // This loop auto generates the highest/lowest number depending on the game version.
+  var index = 0;
+  var maxIndex = 0;
+  var maxNum = 0;
+  var tempArray = diceArray;
+  while (autoCombinedNum.toString().length <= diceArray.length + 1) {
+    if (gameVersion == "highest") {
+      while (index < tempArray.length) {
+        if (tempArray[index] > maxNum) {
+          maxNum = tempArray[index];
+          maxIndex = index;
+          index += 1;
+        } else {
+          index += 1;
+        }
+      }
+      autoCombinedNum = autoCombinedNum * 10 + maxNum;
+      tempArray.splice(maxIndex, 1);
+      maxNum = 0;
+      index = 0;
+    } else {
+      maxNum = 7;
+      while (index < tempArray.length) {
+        if (tempArray[index] < maxNum) {
+          maxNum = tempArray[index];
+          maxIndex = index;
+          index += 1;
+        } else {
+          index += 1;
+        }
+      }
+      autoCombinedNum = autoCombinedNum * 10 + maxNum;
+      tempArray.splice(maxIndex, 1);
+      maxNum = 7;
+      index = 0;
+    }
+  }
```

Comment:
Since you've started using for loops. It might be cleaner to use them here instead of while :) 
Same goes for the rest of the while loops